lambda/functions/schedule/check-black-list/handler.ts

```diff
@@ -0,0 +1,65 @@
+import { findAllBlackListItems, deleteEmailsFromBlacklist } from '../../../shared/services/dynamo.service';
+import {sendPayloadToSqs} from "../../../shared/services/sqs.service";
+import {BlackWhiteListEntity} from "../../../shared/interfaces";
+import {checkEmailDomain} from "../../../shared/utils";
+import {saveTSMessage} from "../../../shared/services/timestream.service";
+import {ValidationLogNames} from "../../../shared/enums";
+
+const oneWeekInMilliseconds = 7 * 24 * 60 * 60 * 1000;
+
+export const handler = async (): Promise<void> => {
+  try {
+    const oneWeekAgo = new Date(+(new Date()) - oneWeekInMilliseconds).toISOString();
+    const blackListItems = await findAllBlackListItems({
+      FilterExpression: 'createdAt < :oneWeekAgo',
+      ExpressionAttributeValues: {
+        ':oneWeekAgo': oneWeekAgo,
+      },
+    });
+
+    // dividing emails in the rehabilitate and check groups
+    const [itemsNeedToRehabilitate, itemsNeedToCheck] = blackListItems.reduce(([a, b], item) => {
+      const rehabilitatedEmailDomains = (process.env.REHABILITATED_EMAIL_DOMAINS!).split(',');
+      const isRehabilitatedEmail = checkEmailDomain(rehabilitatedEmailDomains, item.email);
+      if (isRehabilitatedEmail) {
+        return [[...a, item], b];
+      }
+      const bannedEmailDomains = (process.env.BANNED_EMAIL_DOMAINS!).split(',');
+      const isBannedEmail = checkEmailDomain(bannedEmailDomains, item.email);
+      if (isBannedEmail) {
+        return [a, b];
+      }
+
+      return [a, [...b, item]];
+    }, [[] as BlackWhiteListEntity[], [] as BlackWhiteListEntity[]]);
+
+    const rehabilitatedEmails = itemsNeedToRehabilitate.map(({ email }) => email);
+
+    // removing items from the blackList to recheck
+    await deleteEmailsFromBlacklist([...rehabilitatedEmails, ...itemsNeedToCheck.map((item) => item.email)]);
+
+    // send emails to the SQS to recheck the validation
+    for (const record of itemsNeedToCheck) {
+      try {
+        const { requestId, email, score} = record;
+        const sqsMessageId = await sendPayloadToSqs(requestId, email);
```

Comment:
@G-Vadim There is a logic issue, we will loop the same banned email through the pipe every 2 days, and in the future, we will have a lot of prohibited emails that will run into the validation pipe infinitely. 
The suggestion here is to use timestamps or attempts to prevent this loop, like getting 2 failed attempts equals permanently banned. also, let's have some geometric progress on getting email: 1 try - 2 days, 2 try - 2 weeks, 3 - 1 month  (permanent ban)

Reply:
I added a logic that checks these variants of rechecking:
1. 0 attempts(new record) - check after 1 week
2. 1 attempt - check after 2 weeks
3. 2 attempts - check after 1 month
4. 3 attempts - permanent ban